src/os/bluestore/NVMEDevice.cc

```diff
@@ -1075,13 +1084,12 @@ int NVMEDevice::read(uint64_t off, uint64_t len, bufferlist *pbl,
     t->copy_to_buf(p, 0, t->len);
   };
   ++ioc->num_running;
+  assert(ioc->num_running == 1);
   if(queue_id == -1)
     queue_id = ceph_gettid();
   driver->get_queue(queue_id)->queue_task(t);
 
-  while(t->return_code > 0) {
-    t->io_wait();
-  }
+  ioc->aio_wait();
```

Comment:
I prefer t->io_wait, since one ioc may contains several tasks.

Reply:
same reason. Moreover, we need to use the same interface to handle read and write. Also for the tasks, the number is 1 (if it is not split).